pkg/node/uevent.go

```diff
@@ -69,13 +70,35 @@ func (handler *ueventHandler) syncDrive(
 
 		delete(devices, device.Name)
 
-		var updated bool
-		if drive, updated = updateDriveProperties(drive, device); updated {
-			if _, err := handler.directCSIClient.DirectV1beta3().DirectCSIDrives().Update(
+		var updated, nameChanged bool
+		if drive, updated, nameChanged = updateDriveProperties(drive, device); updated {
+			_, err := handler.directCSIClient.DirectV1beta3().DirectCSIDrives().Update(
 				ctx, &drive, metav1.UpdateOptions{TypeMeta: utils.DirectCSIDriveTypeMeta()},
-			); err != nil {
+			)
+			if err != nil {
 				klog.ErrorS(err, "unable to update drive by "+matchName, "Path", drive.Status.Path, "device.Name", device.Name)
 			}
+
+			if err == nil && nameChanged {
+				volumeInterface := handler.directCSIClient.DirectV1beta3().DirectCSIVolumes()
+				for _, finalizer := range drive.GetFinalizers() {
+					if !strings.HasPrefix(finalizer, directcsi.DirectCSIDriveFinalizerPrefix) {
+						continue
+					}
+
+					volumeName := strings.TrimPrefix(finalizer, directcsi.DirectCSIDriveFinalizerPrefix)
+					volume, err := volumeInterface.Get(ctx, volumeName, metav1.GetOptions{TypeMeta: utils.DirectCSIVolumeTypeMeta()})
+					if err != nil {
```

Comment:
can we retry on conflict here? There could be a scenario where drive path gets updated successfully but volume update fails due to update conflict

Reply:
Done